packages/prover/src/proof_provider/proof_provider.ts

```diff
@@ -0,0 +1,182 @@
+import {Api, getClient} from "@lodestar/api/beacon";
+import {ChainForkConfig, createChainForkConfig} from "@lodestar/config";
+import {networksChainConfig} from "@lodestar/config/networks";
+import {Lightclient, LightclientEvent, RunStatusCode} from "@lodestar/light-client";
+import {LightClientRestTransport} from "@lodestar/light-client/transport";
+import {isForkExecution} from "@lodestar/params";
+import {allForks, capella, ssz} from "@lodestar/types";
+import {LCTransport, RootProviderInitOptions} from "../interfaces.js";
+import {assertLightClient} from "../utils/assertion.js";
+import {
+  getExecutionPayloads,
+  getGenesisData,
+  getSyncCheckpoint,
+  getUnFinalizedRangeForPayloads,
+} from "../utils/consensus.js";
+import {PayloadStore} from "./payload_store.js";
+
+type RootProviderOptions = Omit<RootProviderInitOptions, "transport"> & {
+  transport: LightClientRestTransport;
+  api: Api;
+  config: ChainForkConfig;
+};
+
+export class ProofProvider {
+  private store: PayloadStore;
+
+  private readyPromise?: Promise<void>;
+  lightClient?: Lightclient;
+
+  constructor(private opts: RootProviderOptions) {
+    this.store = new PayloadStore({api: opts.api});
+  }
+
+  async waitToBeReady(): Promise<void> {
+    return this.readyPromise;
+  }
+
+  static init(opts: RootProviderInitOptions): ProofProvider {
+    if (opts.transport === LCTransport.P2P) {
+      throw new Error("P2P mode not supported yet");
+    }
+
+    const config = createChainForkConfig(networksChainConfig[opts.network]);
+    const api = getClient({urls: opts.urls}, {config});
+    const transport = new LightClientRestTransport(api);
+
+    const provider = new ProofProvider({
+      ...opts,
+      config,
+      api,
+      transport,
+    });
+
+    provider.readyPromise = provider.sync(opts.wsCheckpoint);
```

Comment:
add a .catch to `provider.sync(...).catch(e=>{})` to make sure it doesn't throw unhandled exceptions